### PR TITLE
Feature 1650 update hovmoeller

### LINKED ISSUE #1650: Update Hovmoeller Use case to use updated Hovmoeller plotting
*Replace italics below with details for this issue.*

## Describe the Task ##
The original METplotpy hovmoeller plot was not configurable for users who are running from the command line (and do not have permission to modify the code or to indicate where to save the output plot).  Additionally, this code did not follow the paradigm that other METplotpy plots employed: default config file and custom config file.  Instead, it was a mix of a "default" config file that couldn't be modified by some users (ie those that don't have permission to modify the source code that is installed in a common area) with command line arguments. 

### Time Estimate ###
*Estimate the amount of work required here.*
~2 days 

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Update the hovmoeller.yaml config file to match the updated config file structure*
- [ ] *Update/simplify the hovmoeller.py code, the METplotpy/metplotpy/plots/hovmoeller.py code has support for a custom configuration file to override the settings in the default.*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [x] Select **engineer(s)** Minna Win, Hank Fisher
- [x] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority** Assigned High priority because changes to hovmoeller code in METplotpy will be in the beta1 release
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by bikegeek**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Re-created the Hovmoeller plot in the METplus use case following these instructions:
https://metplus.readthedocs.io/en/feature_1650_update_hovmoeller/generated/model_applications/s2s/UserScript_obsPrecip_obsOnly_Hovmoeller.html#sphx-glr-generated-model-applications-s2s-userscript-obsprecip-obsonly-hovmoeller-py

to generate this plot, which uses the same settings employed in the original hovmoeller.yaml config file under the $METPLUS_BASE/METplus/parm/use_cases/model_applications/s2s/UserScript_obsPrecip_obsOnly_Hovmoeller dirctory:

erai_precip2

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Following the use case instructions:
https://metplus.readthedocs.io/en/feature_1650_update_hovmoeller/generated/model_applications/s2s/UserScript_obsPrecip_obsOnly_Hovmoeller.html#sphx-glr-generated-model-applications-s2s-userscript-obsprecip-obsonly-hovmoeller-py

-  on host 'seneca', use the conda environment _/home/met_test/.conda/envs/metplotpy_py38_6_
-  set the METPLUS_BASE, MET_BUILD_BASE, METPLUS_DATA environment variables:
    -  METPLUS_BASE=/d1/personal/mwin/testing_hovmoeller <---Change this to your workspace
    -  MET_BUILD_BASE=/usr/local/met-10.0.0
    -  METPLUS_DATA=/d1/projects/METplus/METplus_Data
    
- in the $METPLUS_BASE directory, clone the following source code:
    -  METcalcpy (**develop** branch)
    -  METplotpy (**feature_213_hovmoeller_configurable** branch)
    -  METplus (**feature_1650_update_hovmoeller** branch)
   
- python $METPLUS_BASE/METplus/ush/run_metplus.py $METPLUS_BASE/METplus/parm/use_cases/model_applications/s2s/UserScript_obsPrecip_obsOnly_Hovmoeller.conf  $METPLUS_BASE/METplus/parm/use_cases/model_applications/s2s/user_system.conf

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[NA]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [ ] Please complete this pull request review by **Tuesday June 21**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

**Review by hankenstein2 — APPROVED** [review]
Everything looks good, recreated successfully on seneca. The automated tests that did not pass are known and ok.

**Review by georgemccabe — APPROVED** [review]
I made a few minor changes to get the use case to succeed in the automated tests. The new input file path was referencing an environment variable METPLUS_DATA that is not set in the use case conf. Also, the YAML file path defined in the use case conf was not being read, so I modified the user script to use that value instead of using a hard-coded path to the yaml file in that directory. We also wanted to wait until the pull request for METplotpy was merged to test that those changes play nicely with these changes. Everything is passing now, so I approve.

**Pull request merged**
